pkg/controller/shared_state.go

```diff
@@ -1564,3 +1571,60 @@ func (c *SharedState) getNodesTopologySelector(ctx context.Context, nodes []v1.N
 	}
 	return topologySelector
 }
+
+func (s *SharedState) addNodeToAvailableAttachmentsMap(ctx context.Context, node string) {
```

Comment:
Where this function is called, it seems like you have the `Node` object already available. You should pass it to this function instead of getting it again.